src/ApiService/ApiService/onefuzzlib/Metrics.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.OneFuzz.Service.OneFuzzLib.Orm;
+
+namespace Microsoft.OneFuzz.Service {
+
+    public record CustomMetric
+    (
+        string Name,
+        int Value,
+        DateTime CreatedAt,
+        BaseMetric CustomDimensions
+    );
+
+
+    public interface IMetrics {
+        Async.Task SendMetric(int metricValue, BaseMetric customDimensions);
+
+        void LogMetric(BaseMetric metric);
+    }
+
+    public class Metrics : IMetrics {
+        private readonly ILogTracer _log;
+        private readonly IOnefuzzContext _context;
+        private readonly JsonSerializerOptions _options;
+
+        public Metrics(ILogTracer log, IOnefuzzContext context) {
+            _context = context;
+            _log = log;
+            _options = new JsonSerializerOptions(EntityConverter.GetJsonSerializerOptions()) {
+                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull
+            };
+            _options.Converters.Add(new RemoveUserInfo());
+        }
+
+        public async Async.Task QueueCustomMetric(MetricMessage message) {
```

Comment:
I think we should treat these metrics as best-effort here, so if we try to enqueue the metric and it fails we should capture and swallow the exception, in order that we not disrupt the ordinary flow of the code sending the metric.

A more advanced implementation would be that this synchronously adds the event to an in-memory queue which is added to the storage queue by a background task, so that "sending" a metric is instantaneous and cannot fail from the POV of the calling code. But this could be left for later.